tests/generators/schema/person-test.php

```diff
@@ -0,0 +1,610 @@
+	 *
+	 * @covers ::__construct
+	 * @covers ::is_needed
+	 */
+	public function test_is_not_needed() {
+		$this->context->site_represents        = 'organization';
+		$this->context->indexable->object_type = 'post';
+
+		$this->assertFalse( $this->instance->is_needed( $this->context ) );
+	}
+
+	/**
+	 * Sets the tests for determine_user_id.
+	 *
+	 * @param string $scenario The scenario to set.
+	 */
+	protected function expects_for_determine_user_id( $scenario = 'default' ) {
+		$user_id = $this->context->site_user_id;
+
+		switch ( $scenario ) {
+			case 'false':
+				$user_id = false;
+				break;
+			case 'zero':
+				$user_id = 0;
+				break;
+		}
+
+		Filters\expectApplied( 'wpseo_schema_person_user_id' )
+			->once()
+			->with( $this->context->site_user_id )
+			->andReturn( $user_id );
+	}
+
+	/**
+	 * Sets the tests for get_userdata inside build_person_data.
+	 *
+	 * @param object|false $user_data The user data get_userdata returns. An object representing WP_User or false.
+	 */
+	protected function expects_for_get_userdata( $user_data ) {
+		Functions\expect( 'get_userdata' )
+			->once()
+			->with( $this->context->site_user_id )
+			->andReturn( $user_data );
+		$this->id->expects( 'get_user_schema_id' )
+			->once()
+			->with( $this->context->site_user_id, $this->context )
+			->andReturn( 'person_id' );
+
+		// No more tests needed when there is no user data.
+		if ( $user_data === false ) {
+			return;
+		}
+
+		$this->html->expects( 'smart_strip_tags' )
+			->once()
+			->with( $user_data->display_name )
+			->andReturn( $user_data->display_name );
+
+		if ( empty( $user_data->description ) ) {
+			$this->html->expects( 'smart_strip_tags' )
+				->never()
+				->with( $user_data->description );
+			return;
+		}
+
+		$this->html->expects( 'smart_strip_tags' )
+			->once()
+			->with( $user_data->description )
+			->andReturn( $user_data->description );
+	}
+
+	/**
+	 * Sets the tests for get_social_profiles.
+	 *
+	 * @param string[] $social_profiles The social profiles after the `wpseo_schema_person_social_profiles` filter.
+	 */
+	protected function expects_for_social_profiles( $social_profiles ) {
+		Filters\expectApplied( 'wpseo_schema_person_social_profiles' )
+			->once()
+			->with( $this->social_profiles, $this->context->site_user_id )
+			->andReturn( $social_profiles );
+
+		if ( empty( $social_profiles ) || ! \is_array( $social_profiles ) ) {
+			Functions\expect( 'get_the_author_meta' )
+				->never();
+
+			return;
+		}
+
+		// Tests for the method `url_social_site`.
+		foreach ( $social_profiles as $social_profile ) {
+			if ( ! \is_string( $social_profile ) ) {
+				Functions\expect( 'get_the_author_meta' )
+					->never()
+					->with( $social_profile, $this->context->site_user_id );
+
+				continue;
+			}
+
+			Functions\expect( 'get_the_author_meta' )
+				->once()
+				->with( $social_profile, $this->context->site_user_id )
+				->andReturn( 'https://example.com/social/' . $social_profile );
```

Comment:
I personally prefer an `if-else` construct, instead of an `if-continue` construct. In my opinion it is more readable.

Reply:
You can also save the expectation in an intermediate variable and add to it, instead of supplying the expectation in one go:
```php
		foreach ( $social_profiles as $social_profile ) {
            // We expect `get_the_author_meta` to be called with the right arguments.
			$expectation = Functions\expect( 'get_the_author_meta' )
				->with( $social_profile, $this->context->site_user_id );

			if ( ! \is_string( $social_profile ) ) {
                // It should never be called when the social profile is not a string.
				$expectation->never();
			}
			else {
                // It should be called once per social profile and return the right URL.
				$expectation
					->once()
					->andReturn( 'https://example.com/social/' . $social_profile );
			}
		}
```